server/bin/pbench-backup-tarballs.py

```diff
@@ -240,18 +242,26 @@ def backup_to_s3(s3_obj, logger, controller_path, controller, tb, tar, resultnam
     except Exception:
         s3_md5 = None
     else:
-        s3_md5 = obj['ETag'].strip("\"")
+        if obj['Metadata']:
```

Comment:
What if there is no metadata key on the obj? You might get an unhandled KeyError exception.
Would use the get method instead which returns None if not there, which will then evaluate to False. 

obj.get('Metadata')

Reply:
Now I see from the s3backup/__init__.py that the Metadata key would always be there, still if an empty dict is return from the get, this will also evaluate to False.

Reply:
Seems best to be defensive here, and verify we got back a `dict` with a `Metadata` member.

Reply:
The code has been moved into a function in the s3backup module and appropriate defensive measures have been added.